src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java

```diff
@@ -0,0 +1,110 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.QueryResultIterable;
+
+/**
+ * A class for implementing the platform dao.
+ */
+public class PlatformReleaseDaoDatastore implements PlatformReleaseDao {
+
+    public static final DatastoreService DATASTORE = DatastoreServiceFactory.getDatastoreService();
+
+    // Get a list of platforms from the datastore
+    public List<Platform> getPlatforms(){
+        List<Platform> platforms = new ArrayList<>();
+        
+        Query query = new Query("Platform");
+        PreparedQuery results = DATASTORE.prepare(query);
+        QueryResultIterable<Entity> entities = results.asQueryResultIterable();
+
+        for (Entity result : entities) {
+            String name = (String) result.getProperty("name");
+            Platform platform = Platform.get(name);
+            platforms.add(platform);
+        }
+        return platforms;  
+    }
+
+    // Get a list releases of a platform from the Datastore
+    public List<Release> getPlatformReleases(Platform platform) {
+        List<Release> releases = new ArrayList<>();
+        Query query = new Query("Release");
+    
+        PreparedQuery preparedQuery = DATASTORE.prepare(query);
+        QueryResultIterable<Entity> releaseQuery = preparedQuery.asQueryResultIterable();
+
+        for (Entity releaseEntity : releaseQuery) {
+            String name = (String) releaseEntity.getProperty("platform");
+            Platform releasePlatform = Platform.get(name);
+            String releaseManager = (String) releaseEntity.getProperty("releaseManager");
+            String releaseName = (String) releaseEntity.getProperty("releaseName");
+            String buganizerHotlistLink = (String) releaseEntity.getProperty("buganizerHotlistLink");
+            Instant launchDate = Instant.ofEpochMilli((Long) releaseEntity.getProperty("launchDate"));
+            Instant launchCalDeadline = Instant.ofEpochMilli((Long) releaseEntity.getProperty("launchCalDeadline"));
+            Instant codeFreezeTime = Instant.ofEpochMilli((Long) releaseEntity.getProperty("codeFreezeTime"));
+           
+            Release release = Release.newBuilder()
+                .platform(releasePlatform)
+                .releaseName(releaseName)
+                .releaseManager(releaseManager)
+                .buganizerHotlistLink(buganizerHotlistLink)
+                .launchCalDeadline(launchCalDeadline)
+                .codeFreezeTime(codeFreezeTime)
+                .launchDate(launchDate).build();
+            if (releasePlatform == platform) {
```

Comment:
So there is a difference between the two methods because in getSDKs you know the platform and sdk, but in this case the only value in the platform. The makePropertyFilter requires a key but all the sdks have different keys.

Reply:
I have two functions. One is getSDK and one is getSDKs, the one you saw was the getSDK however I am referring to the other one on line 70